backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/modelvalidator/C_Order.java

```diff
@@ -139,4 +171,128 @@ else if (X_C_DocType.DOCSUBTYPE_PrepayOrder.equals(dt.getDocSubType())
 
 		return true;
 	}
+
+	@ModelChange(timings = { ModelValidator.TYPE_AFTER_CHANGE },
+			ifColumnsChanged = { I_C_Order.COLUMNNAME_UserElementString1,
+					I_C_Order.COLUMNNAME_UserElementString2,
+					I_C_Order.COLUMNNAME_UserElementString3,
+					I_C_Order.COLUMNNAME_UserElementString4,
+					I_C_Order.COLUMNNAME_UserElementString5,
+					I_C_Order.COLUMNNAME_UserElementString6,
+					I_C_Order.COLUMNNAME_UserElementString7 }
+
+	)
+	public void updateInvoiceCandidatesUserElementStrings(@NonNull final I_C_Order order)
+	{
+		final ClientAndOrgId clientAndOrgId = ClientAndOrgId.ofClientAndOrg(order.getAD_Client_ID(), order.getAD_Org_ID());
+
+		final boolean use1_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement1_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+		final boolean use2_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement2_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+		final boolean use3_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement3_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+		final boolean use4_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement4_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+		final boolean use5_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement5_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+		final boolean use6_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement6_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+		final boolean use7_orderToIc = sysConfigBL.getBooleanValue(SYS_CONFIG_UserStringElement7_DirectUpdateOrderToInvoiceCandidate, false, clientAndOrgId);
+
+		final List<I_C_Invoice_Candidate> invoiceCandidates = invoiceCandDAO.retrieveInvoiceCandidatesForOrderId(OrderId.ofRepoId(order.getC_Order_ID()));
+		for (final I_C_Invoice_Candidate invoiceCandidate : invoiceCandidates)
+		{
+			if (use1_orderToIc && !Objects.equals(invoiceCandidate.getUserElementString1(), order.getUserElementString1()))
```

Comment:
IMHO you don't need `&& !Objects.equals(invoiceCandidate.getUserElementString1(), order.getUserElementString1())`